### PR TITLE
test: Verify (de)serialization of nan, inf, -inf

### PR DESCRIPTION
**Pull request by riddell-stan**
Verify that nan, inf, and -inf values are correctly
serialized.

### CODE REVIEW

File: tests/test_nan_inf.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import math
+
+import pytest
+
+import helpers
+
+program_code = """
+    parameters {
+      real eta;
+      real positive_infinity;
```
Comment (ahartikainen) [inline tests/test_nan_inf.py:11]: why are these defined in parameters block? Aren't these special functions?

Comment (riddell-stan) [reply tests/test_nan_inf.py:11]: I don't know. It seems like this is how it is done in the Stan language. See https://github.com/stan-dev/stan/blob/develop/src/test/test-models/good/nullary-unconflicted.stan

If there's a better way, I'm happy to change things. This seems to work.

Comment (ahartikainen) [reply tests/test_nan_inf.py:11]: I was just reading from here https://mc-stan.org/docs/2_25/functions-reference/special-values.html

**Review by ahartikainen — APPROVED** [review]

File: tests/test_nan_inf.py (hunk 1 above)
Comment (riddell-stan) [reply tests/test_nan_inf.py:11]: I'm really short on time and I'd like to get httpstan and pystan fixed. If you think there's a better way to write this test---and there certainly may be a better way---perhaps we could discuss it in a new issue?

**Pull request merged**